server/lib/nl/nl_data_spec_next.py

```diff
@@ -498,6 +498,26 @@ def _places_from_context(uttr: Utterance) -> List[Place]:
   return ans
 
 
+# Computes a list of place lists, where each inner list is a candidate for place
+# comparison.  The outer list is ordered from most prefered candidates to least.
+#
+# The logic for determining a single place comparison candidate is as follows:
+# 1. If an utterance has multiple places, then that is a candidate list.
+# 2. If an utterance has a single place, then we walk up prior utterances until
+#    we find an utterance with at least one place.  The candidate list is
+#    the former single place plus the latter from prior utterance.
+#
+# For example, suppose the query sequence is as follows:
+#   [tell me about san jose] -> [how is auto theft in fremont] -> [compare with palo alto]
+#
+# In this case, the ordered candidates will be:
+#   [[palo alto, fremont], [palo alto, san jose], [fremont, san jose]]
```

Comment:
So we will show comparison for all the 3 combinations? Not sure if that will be too much, esp if there are more context places before.

Reply:
No, the caller will loop through each combo, and stop when the first one generated any charts.  (This is part of the general pattern of trying to show something when data doesn't exist for certain places)

Reply:
Ok, so here will fill as much as possible, now I see the pattern..